src/Controls/tests/BindingSourceGen.UnitTests/IntegrationTests.cs

```diff
@@ -1434,4 +1434,505 @@ internal static partial class GeneratedBindingInterceptors
+                        Column = column;
+                    }
+
+                    public string FilePath { get; }
+                    public int Line { get; }
+                    public int Column { get; }
+                }
+            }
+
+            namespace Microsoft.Maui.Controls.Generated
+            {
+                using System;
+                using System.CodeDom.Compiler;
+                using System.Runtime.CompilerServices;
+                using Microsoft.Maui.Controls.Internals;
+
+                internal static partial class GeneratedBindingInterceptors
+                {
+
+                    {{BindingCodeWriter.GeneratedCodeAttribute}}
+                    [InterceptsLocationAttribute(@"Path\To\Program.cs", 17, 23)]
+                    public static void SetBinding{{id}}(
+                        this BindableObject bindableObject,
+                        BindableProperty bindableProperty,
+                        Func<global::MyNamespace.MySourceClass, string> getter,
+                        BindingMode mode = BindingMode.Default,
+                        IValueConverter? converter = null,
+                        object? converterParameter = null,
+                        string? stringFormat = null,
+                        object? source = null,
+                        object? fallbackValue = null,
+                        object? targetNullValue = null)
+                    {
+                        Action<global::MyNamespace.MySourceClass, string>? setter = null;
+                        if (ShouldUseSetter(mode, bindableProperty))
+                        {
+                            throw new InvalidOperationException("Cannot set value on the source object.");
+                        }
+
+                        var binding = new TypedBinding<global::MyNamespace.MySourceClass, string>(
+                            getter: source => (getter(source), true),
+                            setter,
+                            handlers: new Tuple<Func<global::MyNamespace.MySourceClass, object?>, string>[]
+                            {
+                                new(static source => source, "Text"),
+                            })
+                        {
+                            Mode = mode,
+                            Converter = converter,
+                            ConverterParameter = converterParameter,
+                            StringFormat = stringFormat,
+                            Source = source,
+                            FallbackValue = fallbackValue,
+                            TargetNullValue = targetNullValue
+                        };
+
+                        bindableObject.SetBinding(bindableProperty, binding);
+                    }
+
+                    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "get_Text")]
+                    private static extern string GetUnsafeProperty{{id}}Text(global::MyNamespace.MySourceClass source);
```

Comment:
Just a side note: we don't really need the unsafe getter when it's the LAST property in the path (because we only ever call it in the handlers array), but I don't think we need to spend time optimizing this at the moment.